### PR TITLE
[v2.9] Integration tests

### PR DESCRIPTION
**Pull request by rafaelbreno**
### Automated Testing

* Test types added/modified:
    * Unit
    * Integration (Go Framework)
    * Integration (v2prov Framework)
    * Validation (Go Framework)
    * Other - Explain: _EXPLAIN_
    * None
    * _REMOVE NOT APPLICABLE BULLET POINTS ABOVE_
* If "None" - Reason: _EXPLAIN THE REASON_
  
* If "None" - GH Issue/PR: _LINK TO GH ISSUE/PR TO ADD TESTS_

Summary: _TODO_

### Regressions Considerations

_TODO_

Existing / newly added automated tests that provide evidence there are no regressions:
* _TODO_

### CODE REVIEW

File: .github/workflows/integration-tests.yml [hunk 1]
```diff
@@ -0,0 +1,360 @@
...
+            cp /bin/ip /host-bin/ &&
+            cp /bin/ipset /host-bin/ &&
+            cp /bin/k3s /host-bin/ &&
+            cp /bin/losetup /host-bin/ &&
+            cp /bin/pigz /host-bin/ &&
+            cp /bin/runc /host-bin/ &&
+            cp /bin/which /host-bin/ &&
+            cp /bin/aux/xtables-legacy-multi /host-bin/"
+
+      #- name: List copied binaries
```
Comment (tashima42) [inline .github/workflows/integration-tests.yml:151]: Can you remove this comment?

File: .github/workflows/integration-tests.yml [hunk 2]
```diff
@@ -0,0 +1,360 @@
...
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Clean up Docker
+        run: |
+          docker system prune -af
+          docker volume prune -f
+
+      - name: Clean up temporary files
+        run: rm -rf /path/to/temp/files
```
Comment (tashima42) [inline .github/workflows/integration-tests.yml:308]: I think this path is not the intended one

File: scripts/gha-build [hunk 3]
```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+set -e
+
+cd $(dirname $0)
+
+./validate
```
Comment (tashima42) [inline scripts/gha-build:6]: We're already running validate in other workflows, I don't think we need to run it here

File: tests/integration/tox.ini [hunk 4]
```diff
@@ -2,7 +2,7 @@
 envlist = flake8,rancher
 
 [testenv]
-basepython = python3.11
+basepython = python3.10
```
Comment (tashima42) [inline tests/integration/tox.ini:5]: Is this intentional?

Comment (rafaelbreno) [reply tests/integration/tox.ini:5]: Yes, it's, GitHub runners comes with default python version 3.10, and it doesn't allow use to update the version to a newer one.

**Comment by igomez06** [comment]
Closing in favor of:
https://github.com/rancher/rancher/pull/46036

**Pull request closed without merging**